### PR TITLE
Fix bug in filtering, cleanup

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

This PR fixes a bug results in the `top_n` filtering options not being applied in `system/process`, as well as refactors a bunch of components to make the new linter happy.

## Why is it important?

This is an actual bug. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Pull down and build, make sure the `include_top_n` flags work in metricbeat.

### CODE REVIEW

File: libbeat/metric/system/process/helpers.go [hunk 1]
```diff
@@ -79,10 +79,14 @@ func GetProcCPUPercentage(s0, s1 ProcState) ProcState {
 	}
 
 	timeDelta := s1.SampleTime.Sub(s0.SampleTime)
-	timeDeltaMillis := timeDelta / time.Millisecond
+	timeDeltaMillis := float64(timeDelta / time.Millisecond)
 	totalCPUDeltaMillis := int64(s1.CPU.Total.Ticks.ValueOr(0) - s0.CPU.Total.Ticks.ValueOr(0))
 
-	pct := float64(totalCPUDeltaMillis) / float64(timeDeltaMillis)
+	// In theory this can only happen if the time delta is 0, which is unlikely but possible.
+	if timeDeltaMillis == 0 {
```
Comment (cmacknz) [inline libbeat/metric/system/process/helpers.go:86]: You are doing this comparison against a float now, which leaves it open to floating point precisions issues comparing against an exact value.

File: libbeat/metric/system/process/process.go [hunk 2]
```diff
@@ -203,13 +202,13 @@ func (procStats *Stats) Get() ([]mapstr.M, []mapstr.M, error) {
...
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "error gathering PIDs")
+		return nil, nil, fmt.Errorf("error gathering PIDs: %w", err)
 	}
 	// We use this to track processes over time.
 	procStats.ProcsMap = pidMap
 
 	// filter the process list that will be passed down to users
-	procStats.includeTopProcesses(plist)
+	plist = procStats.includeTopProcesses(plist)
```
Comment (cmacknz) [inline libbeat/metric/system/process/process.go:211]: Is this the primary bug?

File: libbeat/metric/system/process/process_test.go [hunk 3]
```diff
@@ -61,9 +58,54 @@ func TestGetOne(t *testing.T) {
...
+
+	time.Sleep(time.Second * 2)
+
+	procData, _, err := testConfig.Get()
+	assert.NoError(t, err, "GetOne")
+
+	t.Logf("Proc: %s", procData[0].StringToPrint())
+}
+
+func TestFilter(t *testing.T) {
```
Comment (cmacknz) [inline libbeat/metric/system/process/process_test.go:72]: This was the missing test from the looks of things?

File: libbeat/metric/system/process/process.go (hunk 2 above)
Comment (fearful-symmetry) [reply libbeat/metric/system/process/process.go:211]: yup.

File: libbeat/metric/system/process/process_test.go (hunk 3 above)
Comment (fearful-symmetry) [reply libbeat/metric/system/process/process_test.go:72]: Yah, added that.

**Comment by fearful-symmetry** [comment]
@kvch have we solved the "typecheck failing on stuff that has cgo" issue yet?

**Pull request merged**